projects/js-packages/connection/components/disconnect-dialog/test/step-disconnect.jsx

```diff
@@ -35,25 +40,66 @@ describe( 'StepDisconnect', () => {
 
 	describe( 'When the disconnect button is clicked', () => {
 		const wrapper = shallow( <StepDisconnect { ...testProps } /> );
-
-		wrapper
-			.find( '.jp-connection__disconnect-dialog__btn-disconnect' )
-			.simulate( 'click', { preventDefault: () => undefined } );
+		const disconnectButton = wrapper.find( '.jp-connection__disconnect-dialog__btn-disconnect' );
 
 		it( 'calls the disconnect callback', () => {
+			disconnectButton.simulate( 'click', { preventDefault: () => undefined } );
+
 			expect( testProps.onDisconnect.called ).to.be.true;
 		} );
+
+		it( 'calls the trackModalClick callback with jetpack_disconnect_dialog_click_disconnect', () => {
+			disconnectButton.simulate( 'click', { preventDefault: () => undefined } );
+
+			expect(
+				testProps.trackModalClick.calledOnceWith( 'jetpack_disconnect_dialog_click_disconnect' )
+			).to.be.true;
+		} );
 	} );
 
 	describe( 'When the dismiss button is clicked', () => {
 		const wrapper = shallow( <StepDisconnect { ...testProps } /> );
-
-		wrapper
-			.find( '.jp-connection__disconnect-dialog__btn-dismiss' )
-			.simulate( 'click', { preventDefault: () => undefined } );
+		const dismissBtn = wrapper.find( '.jp-connection__disconnect-dialog__btn-dismiss' );
 
 		it( 'calls the closeModal callback', () => {
+			dismissBtn.simulate( 'click', { preventDefault: () => undefined } );
+
 			expect( testProps.closeModal.called ).to.be.true;
 		} );
+
+		it( 'calls the trackModalClick callback with jetpack_disconnect_dialog_click_stay_connected', () => {
+			dismissBtn.simulate( 'click', { preventDefault: () => undefined } );
+
+			expect(
+				testProps.trackModalClick.calledOnceWith( 'jetpack_disconnect_dialog_click_stay_connected' )
+			).to.be.true;
+		} );
+	} );
+
+	describe( 'When help links are clicked', () => {
+		const trackEvents = [
+			'jetpack_disconnect_dialog_click_learn_about',
+			'jetpack_disconnect_dialog_click_support',
+		];
+		const wrapper = shallow( <StepDisconnect { ...testProps } /> );
+		const links = wrapper.find( '.jp-connection__disconnect-dialog__link' );
+		let trackedTotal = 0;
+
+		links.forEach( ( link, i ) => {
+			link.simulate( 'click', { preventDefault: () => undefined } );
+
+			const clickTrackValue = testProps.trackModalClick.getCall( i ).args[ 0 ];
+
+			it( `should track ${ clickTrackValue }`, async () => {
+				const valueIdx = trackEvents.indexOf( clickTrackValue );
+
+				expect( clickTrackValue ).to.equal( trackEvents[ valueIdx ] );
+				trackedTotal++;
+			} );
+		} );
+
+		it( `should track all ${ trackEvents.length } expected events`, () => {
+			expect( trackedTotal ).to.equal( trackEvents.length );
+		} );
```

Comment:
```suggestion
		
		it( 'Calls all expected help link events', () => {
			links.forEach( link => {
				link.simulate( 'click', { preventDefault: () => undefined } );
			} );

			trackEvents.forEach( event_name => {
				expect( testProps.trackModalClick.calledOnceWith( event_name ) );
			} );
		} );
```

This test may not need to be concerned about the order of the events/ callbacks? I think displaying the event names as part of the test name is a nice add, but may be making the test too complex.

Reply:
I like the simplicity of your test 👍 

The order shouldn't matter in my version either. I also wanted to try and catch if people added links without tracking, or tracking links without testing... so I'll leave it as it is.